### PR TITLE
Update code snippets

### PR DESCRIPTION
**Pull request by chrissvo**
Fixes: https://github.com/mgmco/bugs/issues/2338

Changes proposed in this pull request:
 * TSIA

cc @mgmco/developers

### CODE REVIEW

File: _sales_tracking/api.md [hunk 1]
```diff
@@ -2,49 +2,75 @@
...
+    "sale_id": "MGM_SALE_ID",
+    "name": "BUYER_NAME",
+    "email": "BUYER_EMAIL",
+    "state": "confirmed"
+  });
 
-    res.on("data", function (chunk) {
-      chunks.push(chunk);
-    });
+  var xhr = new XMLHttpRequest();
```
Comment (mtimofiiv) [inline _sales_tracking/api.md]: This is frontend JS - there is no `XMLHttpRequest` in the bertbuttend

Comment (chrissvo) [inline _sales_tracking/api.md]: Oeh, oh nee!

File: _sales_tracking/vouchers.md [hunk 2]
```diff
@@ -28,23 +38,46 @@ right_code: |
...
+  OkHttpClient client = new OkHttpClient();
+
+  MediaType mediaType = MediaType.parse("application/json");
+  RequestBody body = RequestBody.create(mediaType, "{\n\t\"source\": \"curl\",\n\t\"api_key\": \"API_KEY\",\n\t\"voucher\": \"VOUCHER_CODE\",\n\t\"name\": \"BUYER_NAME\",\n\t\"email\": \"BUYER_EMAIL\",\n\t\"state\": \"completed\"\n}");
+  Request request = new Request.Builder()
+    .url("https://api.membergetmember.co/sales/status")
+    .post(body)
+    .addHeader("content-type", "application/json")
+    .addHeader("cache-control", "no-cache")
+    .addHeader("postman-token", "96d36bfd-04d3-c17a-4295-fffe0dab11e2")
```
Comment (mtimofiiv) [inline _sales_tracking/vouchers.md]: postman token

File: _sales_tracking/vouchers.md [hunk 3]
```diff
@@ -28,23 +38,46 @@ right_code: |
...
+  url = URI("https://api.membergetmember.co/sales/status")
 
   http = Net::HTTP.new(url.host, url.port)
+  http.use_ssl = true
+  http.verify_mode = OpenSSL::SSL::VERIFY_NONE
 
   request = Net::HTTP::Post.new(url)
+  request["content-type"] = 'application/json'
+  request["cache-control"] = 'no-cache'
+  request["postman-token"] = '6f8b622f-25ff-e45e-2f96-84e8289d8a5a'
```
Comment (mtimofiiv) [inline _sales_tracking/vouchers.md]: postman token!

File: _sales_tracking/vouchers.md [hunk 4]
```diff
@@ -53,37 +86,58 @@ right_code: |
...
-  $client->enqueue($request)->send();
-  $response = $client->getResponse();
-
-  echo $response->getBody();
+  $request = new HttpRequest();
+  $request->setUrl('https://api.membergetmember.co/sales/status');
+  $request->setMethod(HTTP_METH_POST);
+
+  $request->setHeaders(array(
+    'postman-token' => '038efb15-39d7-13b3-f7cf-df37c246c334',
```
Comment (mtimofiiv) [inline _sales_tracking/vouchers.md]: damn you postman!

File: _sales_tracking/vouchers.md [hunk 5]
```diff
@@ -53,37 +86,58 @@ right_code: |
...
+  } catch (HttpException $ex) {
+    echo $ex;
+  }
   ~~~
   {: title="PHP"}
   ~~~csharp
-  var client = new RestClient("https://api.membergetmember.co/sales/status?source=backend_csharp&voucher=VOUCHER&name=NAME&email=EMAIL&api_key=API_KEY");
+  var client = new RestClient("https://api.membergetmember.co/sales/status");
   var request = new RestRequest(Method.POST);
+  request.AddHeader("postman-token", "59a22a7c-1775-368c-cf91-e65528e3556d");
```
Comment (mtimofiiv) [inline _sales_tracking/vouchers.md]: also here

File: _sales_tracking/vouchers.md [hunk 6]
```diff
@@ -53,37 +86,58 @@ right_code: |
...
 
-  conn = http.client.HTTPConnection("api.membergetmember.co")
+  conn = http.client.HTTPSConnection("api.membergetmember.co")
+
+  payload = "{\n\t\"source\": \"curl\",\n\t\"api_key\": \"API_KEY\",\n\t\"voucher\": \"VOUCHER\",\n\t\"name\": \"BUYER_NAME\",\n\t\"email\": \"BUYER_EMAIL\",\n\t\"state\": \"completed\"\n}"
+
+  headers = {
+      'content-type': "application/json",
+      'cache-control': "no-cache",
+      'postman-token': "ab98d8ee-236b-40f5-bc9f-c6e4910a0c18"
```
Comment (mtimofiiv) [inline _sales_tracking/vouchers.md]: penis

File: _sales_tracking/api.md [hunk 7]
```diff
@@ -53,36 +86,58 @@ right_code: |
...
 
-  conn = http.client.HTTPConnection("api.membergetmember.co")
+  conn = http.client.HTTPSConnection("api.membergetmember.co")
+
+  payload = "{\n\t\"source\": \"curl\",\n\t\"api_key\": \"API_KEY\",\n\t\"sale_id\": \"MGM_SALE_ID\",\n\t\"name\": \"BUYER_NAME\",\n\t\"email\": \"BUYER_EMAIL\",\n\t\"state\": \"confirmed\"\n}"
+
+  headers = {
+      'content-type': "application/json",
+      'cache-control': "no-cache",
+      'postman-token': "ab98d8ee-236b-40f5-bc9f-c6e4910a0c18"
```
Comment (mtimofiiv) [inline _sales_tracking/api.md]: AAAAAAAGGGGGGGGHHHHHHHHH

File: _sales_tracking/api.md [hunk 8]
```diff
@@ -28,23 +38,42 @@ right_code: |
...
   req.end();
   ~~~
   {: title="JavaScript" }
   ~~~ java
-  HttpResponse response = Unirest.get("https://api.membergetmember.co/sales/callback?source=backend_java&sale_id=SALE_ID&name=NAME&email=EMAIL")
-  .asString();
+  OkHttpClient client = new OkHttpClient();
+
+  MediaType mediaType = MediaType.parse("application/json");
+  RequestBody body = RequestBody.create(mediaType, "{\n\t\"source\": \"curl\",\n\t\"api_key\": \"API_KEY\",\n\t\"sale_id\": \"MGM_SALE_ID\",\n\t\"name\": \"BUYER_NAME\",\n\t\"email\": \"BUYER_EMAIL\",\n\t\"state\": \"confirmed\"\n}");
```
Comment (harianus) [inline _sales_tracking/api.md]: Isn't there a nice way to do this? This looks really error prone.

File: _sales_tracking/api.md [hunk 9]
```diff
@@ -28,23 +38,42 @@ right_code: |
...
-  url = URI("https://api.membergetmember.co/sales/callback?source=backend_ruby&sale_id=SALE_ID&name=NAME&email=EMAIL")
+  url = URI("https://api.membergetmember.co/sales/status")
 
   http = Net::HTTP.new(url.host, url.port)
+  http.use_ssl = true
+  http.verify_mode = OpenSSL::SSL::VERIFY_NONE
 
-  request = Net::HTTP::Get.new(url)
+  request = Net::HTTP::Post.new(url)
+  request["content-type"] = 'application/json'
```
Comment (harianus) [inline _sales_tracking/api.md]: The rest of this snippet has double quotes. This one not.

File: _sales_tracking/api.md [hunk 10]
```diff
@@ -53,36 +82,52 @@ right_code: |
...
+  }
   ~~~
   {: title="PHP"}
   ~~~csharp
-  var client = new RestClient("https://api.membergetmember.co/sales/callback?source=backend_csharp&sale_id=SALE_ID&name=NAME&email=EMAIL");
-  var request = new RestRequest(Method.GET);
+  var client = new RestClient("https://api.membergetmember.co/sales/status");
+  var request = new RestRequest(Method.POST);
+  request.AddHeader("content-type", "application/json");
+  request.AddParameter("application/json", "{\n\t\"source\": \"curl\",\n\t\"api_key\": \"API_KEY\",\n\t\"sale_id\": \"MGM_SALE_ID\",\n\t\"name\": \"BUYER_NAME\",\n\t\"email\": \"BUYER_EMAIL\",\n\t\"state\": \"confirmed\"\n}", ParameterType.RequestBody);
```
Comment (harianus) [inline _sales_tracking/api.md]: Source should be `backend_csharp`

File: _sales_tracking/api.md [hunk 11]
```diff
@@ -53,36 +82,52 @@ right_code: |
...
   IRestResponse response = client.Execute(request);
   ~~~
   {: title="C#"}
   ~~~python
   import http.client
 
-  conn = http.client.HTTPConnection("api.membergetmember.co")
+  conn = http.client.HTTPSConnection("api.membergetmember.co")
+
+  payload = "{\n\t\"source\": \"curl\",\n\t\"api_key\": \"API_KEY\",\n\t\"sale_id\": \"MGM_SALE_ID\",\n\t\"name\": \"BUYER_NAME\",\n\t\"email\": \"BUYER_EMAIL\",\n\t\"state\": \"confirmed\"\n}"
```
Comment (harianus) [inline _sales_tracking/api.md]: Source should be `backend_python`

File: _sales_tracking/api.md [hunk 12]
```diff
@@ -53,36 +82,52 @@ right_code: |
...
+  $request = new HttpRequest();
+  $request->setUrl('https://api.membergetmember.co/sales/status');
+  $request->setMethod(HTTP_METH_POST);
+
+  $request->setHeaders(array(
+    'content-type' => 'application/json'
+  ));
+
+  $request->setBody('{
+    "source": "curl",
```
Comment (harianus) [inline _sales_tracking/api.md]: Source should be `backend_php`

File: _sales_tracking/api.md [hunk 13]
```diff
@@ -28,23 +38,42 @@ right_code: |
...
+  url = URI("https://api.membergetmember.co/sales/status")
 
   http = Net::HTTP.new(url.host, url.port)
+  http.use_ssl = true
+  http.verify_mode = OpenSSL::SSL::VERIFY_NONE
 
-  request = Net::HTTP::Get.new(url)
+  request = Net::HTTP::Post.new(url)
+  request["content-type"] = 'application/json'
+  request.body = "{\n\t\"source\": \"curl\",\n\t\"api_key\": \"API_KEY\",\n\t\"sale_id\": \"MGM_SALE_ID\",\n\t\"name\": \"BUYER_NAME\",\n\t\"email\": \"BUYER_EMAIL\",\n\t\"state\": \"confirmed\"\n}"
```
Comment (harianus) [inline _sales_tracking/api.md]: Source should be `backend_ruby`

File: _sales_tracking/api.md (hunk 8 above)
Comment (harianus) [inline _sales_tracking/api.md]: Source should be `backend_java`

File: _sales_tracking/api.md [hunk 14]
```diff
@@ -28,23 +38,42 @@ right_code: |
     });
   });
 
+  req.write(JSON.stringify({ source: 'curl',
```
Comment (harianus) [inline _sales_tracking/api.md]: Source should be `backend_nodejs`

File: _sales_tracking/vouchers.md [hunk 15]
```diff
@@ -28,23 +38,42 @@ right_code: |
     });
   });
 
+  req.write(JSON.stringify({ source: 'curl',
```
Comment (harianus) [inline _sales_tracking/vouchers.md]: Source should be `backend_nodejs`

File: _sales_tracking/vouchers.md [hunk 16]
```diff
@@ -28,23 +38,42 @@ right_code: |
...
   req.end();
   ~~~
   {: title="JavaScript" }
   ~~~ java
-  HttpResponse response = Unirest.post("https://api.membergetmember.co/sales/status?source=backend_java&voucher=VOUCHER&name=NAME&email=EMAIL&api_key=API_KEY")
-  .asString();
+  OkHttpClient client = new OkHttpClient();
+
+  MediaType mediaType = MediaType.parse("application/json");
+  RequestBody body = RequestBody.create(mediaType, "{\n\t\"source\": \"curl\",\n\t\"api_key\": \"API_KEY\",\n\t\"voucher\": \"VOUCHER_CODE\",\n\t\"name\": \"BUYER_NAME\",\n\t\"email\": \"BUYER_EMAIL\",\n\t\"state\": \"completed\"\n}");
```
Comment (harianus) [inline _sales_tracking/vouchers.md]: Source should be `backend_java`

File: _sales_tracking/vouchers.md [hunk 17]
```diff
@@ -28,23 +38,42 @@ right_code: |
...
-  url = URI("https://api.membergetmember.co/sales/status?source=backend_ruby&voucher=VOUCHER&name=NAME&email=EMAIL&api_key=API_KEY")
+  url = URI("https://api.membergetmember.co/sales/status")
 
   http = Net::HTTP.new(url.host, url.port)
+  http.use_ssl = true
+  http.verify_mode = OpenSSL::SSL::VERIFY_NONE
 
   request = Net::HTTP::Post.new(url)
+  request["content-type"] = 'application/json'
+  request.body = "{\n\t\"source\": \"curl\",\n\t\"api_key\": \"API_KEY\",\n\t\"voucher\": \"VOUCHER_CODE\",\n\t\"name\": \"BUYER_NAME\",\n\t\"email\": \"BUYER_EMAIL\",\n\t\"state\": \"completed\"\n}"
```
Comment (harianus) [inline _sales_tracking/vouchers.md]: Source should be `backend_ruby`

File: _sales_tracking/vouchers.md [hunk 18]
```diff
@@ -53,37 +82,52 @@ right_code: |
...
+  $request = new HttpRequest();
+  $request->setUrl('https://api.membergetmember.co/sales/status');
+  $request->setMethod(HTTP_METH_POST);
+
+  $request->setHeaders(array(
+    'content-type' => 'application/json'
+  ));
+
+  $request->setBody('{
+    "source": "curl",
```
Comment (harianus) [inline _sales_tracking/vouchers.md]: Source should be `backend_php`

File: _sales_tracking/vouchers.md [hunk 19]
```diff
@@ -53,37 +82,52 @@ right_code: |
...
+    echo $ex;
+  }
   ~~~
   {: title="PHP"}
   ~~~csharp
-  var client = new RestClient("https://api.membergetmember.co/sales/status?source=backend_csharp&voucher=VOUCHER&name=NAME&email=EMAIL&api_key=API_KEY");
+  var client = new RestClient("https://api.membergetmember.co/sales/status");
   var request = new RestRequest(Method.POST);
+  request.AddHeader("content-type", "application/json");
+  request.AddParameter("application/json", "{\n\t\"source\": \"curl\",\n\t\"api_key\": \"API_KEY\",\n\t\"voucher\": \"VOUCHER\",\n\t\"name\": \"BUYER_NAME\",\n\t\"email\": \"BUYER_EMAIL\",\n\t\"state\": \"completed\"\n}", ParameterType.RequestBody);
```
Comment (harianus) [inline _sales_tracking/vouchers.md]: Source should be `backend_csharp`

File: _sales_tracking/vouchers.md [hunk 20]
```diff
@@ -53,37 +82,52 @@ right_code: |
...
   IRestResponse response = client.Execute(request);
   ~~~
   {: title="C#"}
   ~~~python
   import http.client
 
-  conn = http.client.HTTPConnection("api.membergetmember.co")
+  conn = http.client.HTTPSConnection("api.membergetmember.co")
+
+  payload = "{\n\t\"source\": \"curl\",\n\t\"api_key\": \"API_KEY\",\n\t\"voucher\": \"VOUCHER\",\n\t\"name\": \"BUYER_NAME\",\n\t\"email\": \"BUYER_EMAIL\",\n\t\"state\": \"completed\"\n}"
```
Comment (harianus) [inline _sales_tracking/vouchers.md]: Source should be `backend_python`

File: _sales_tracking/api.md (hunk 10 above)
Comment (mtimofiiv) [inline _sales_tracking/api.md]: Also is there no json serializer for this language?

File: _sales_tracking/api.md (hunk 8 above)
Comment (chrissvo) [inline _sales_tracking/api.md]: Google search turns up libraries and plugins. So I guess most JAVA developers use a library like GSON.

File: _sales_tracking/api.md (hunk 10 above)
Comment (chrissvo) [inline _sales_tracking/api.md]: There are plugins, and there seems to be such a thing as the `JavaScriptSerializer`: https://msdn.microsoft.com/en-us/library/system.web.script.serialization.javascriptserializer.aspx

I'm not sure how to test if this actually works though...

**Comment by chrissvo** [comment]
@harianus can you take a second look / merge in?

File: _sales_tracking/api.md [hunk 21]
```diff
@@ -2,17 +2,27 @@
...
+    "name": "BUYER_NAME",
+    "email": "BUYER_EMAIL",
+    "state": "confirmed"
+  }' "https://api.membergetmember.co/sales/status"
   ~~~
-  {: title="Bash"}
+  {: title="cURL"}
   ~~~ javascript
-  var http = require("http");
+  var http = require("https");
```
Comment (harianus) [inline _sales_tracking/api.md]: Should be `https` as well I would say, but not very important

Comment (chrissvo) [inline _sales_tracking/api.md]: @mtimofiiv is this correct?

Comment (mtimofiiv) [inline _sales_tracking/api.md]: The `http` module does not support `https`. So use `https`!

Comment (harianus) [inline _sales_tracking/api.md]: I meant the variable 😚

Comment (chrissvo) [inline _sales_tracking/api.md]: Haha, te laat!

**Pull request merged**